shell/platform/darwin/macos/framework/Source/FlutterTextInputSemanticsObject.h

```diff
@@ -0,0 +1,89 @@
+
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_base.h"
+
+@class FlutterTextField;
+@class FlutterTextInputPlugin;
+
+namespace flutter {
+
+//------------------------------------------------------------------------------
+/// The ax platform node for a text field.
+class FlutterTextPlatformNode : public ui::AXPlatformNodeBase {
+ public:
+  //---------------------------------------------------------------------------
+  /// @brief      Creates a FlutterTextPlatformNode that uses a
+  ///             FlutterTextField as its NativeViewAccessible.
+  /// @param[in]  delegate          The delegate that provides accessibility
+  ///                               data.
+  /// @param[in]  view_controller   The view_controller that is used for querying
+  ///                               the information about FlutterView and
+  ///                               FlutterTextInputPlugin.
+  explicit FlutterTextPlatformNode(FlutterPlatformNodeDelegate* delegate,
+                                   __weak FlutterViewController* view_controller);
+  ~FlutterTextPlatformNode() override;
+
+  //------------------------------------------------------------------------------
+  /// @brief Gets the frame of this platform node relative to the view of
+  ///        FlutterViewController. This is used by the FlutterTextField to get its
+  ///        frame rect because the FlutterTextField is a subview of the
+  ///        FlutterViewController.view.
+  NSRect GetFrameRelativeToControllerView();
+
+  // |ui::AXPlatformNodeBase|
+  gfx::NativeViewAccessible GetNativeViewAccessible() override;
+
+ private:
+  FlutterTextField* native_text_field_;
+  __weak FlutterViewController* view_controller_;
+
+  //------------------------------------------------------------------------------
+  /// @brief Ensures the FlutterTextField is attached to the FlutterView. This
+  ///        method returns true if it is attached after this method excuted.
+  ///        If this method returns false, that means the FlutterTextField cannot
+  ///        attach to the FlutterView. This can happen when the FlutterEngine does
+  ///        not have a FlutterViewController or the FlutterView is not loaded yet.
+  bool EnsureAttachedToView();
+
+  //------------------------------------------------------------------------------
+  /// @brief Detaches the FlutterTextField from the FlutterView if it is not
+  ///        already detached.
+  void EnsureDetachedFromView();
+};
+
+}  // namespace flutter
+
+/**
+ * A native text field that represents the NativeViewAccessible for the FlutterTextPlatformNode
+ *
+ * The NSAccessibility protocol does not provide full support for text editing. This native text
+ * field is used to get around this problem. The FlutterTextPlatformNode creates a hidden
+ * FlutterTextField and makes macOS think that it is interacting with a real native text field.
```

Comment:
nit: it's not really that it 'thinks' it's interacting with a real native text field, it's that it _is_ interacting with a native text field.

Maybe: The FlutterTextPlatformNode creates a hidden FlutterTextField, since VoiceOver only provides full input accessibility support for NSTextField subclasses.